### ISSUE TITLE
CPU allocation on the Kluster doesn't work as expected

### ISSUE THREAD

**Issue by balazsgyenes** [ALRhub/clusterduck#6]
clusterduck detects only 4 CPUs when run with the example config on the Kluster, although 8 CPUs are requested.

Further observations:
- When 4 CPUs are requested, only 2 are auto-detected (which causes a crash with 4 workers). When 16 CPUs are requested, sbatch fails with `Requested node configuration is not available`.
- With an empty allocation (i.e. `salloc`) with the same parameters as the sbatch submission (`--cpus-per-task=8 --mem-per-cpu=15000 --nodes=1 --ntasks-per-node=1`), `psutil` always detects 20 CPUs. I have tried this on multiple nodes (using e.g. `--nodelist=node6`). I get the same result if I manually import and run `create_resource_pools_from_cfg` with a dummy resource config.
- Despite what the documentation says, `psutil.Process().cpu_affinity([])` does not reset the cpu affinity. It seems to have no effect on the CPU affinity whatsoever. This is very easy to verify and seems to behave the same on any system (not just the kluster).

It's currently unclear if this problem is specific to the Kluster or might also happen on other clusters.

**Comment by ScheiklP** [comment]
Huh, github does not show the images for me anymore...

TLDR: It tried several commands on the alex cluster, and the only one that gave me the correct cpu ids was:
`os.sched_getaffinity(1)`

**Comment by nicolas-schreiber** [comment]
Some more observations on Node 6 of the Kluster (with 24 cores) where we asked for 24 cores:
clusterduck lists following cores as available:
```
[2023-10-25 11:11:35,705][clusterduck] Auto-detected the following CPUs: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
```

**Comment by nicolas-schreiber** [comment]
And more observations:

I ran several experiments with following config, specifically using only 1 node and requesting only 1 gpu and 5 of the 24 cpus:

```
defaults:
  - clusterduck_slurm

timeout_min: 2880
partition: gpu
gres: gpu:1

wait_for_completion: false # Needs to be True, otherwise Optuna will not receive the results
cpus_per_task: 5
mem_gb: 16

setup:
  - export TQDM_DISABLE=1

# clusterduck specific options
parallel_runs_per_node: 1
total_runs_per_node: 4

exclude: node[1-5]

resources_config:
  cpu:
  cuda:
  stagger:
    delay: 5


```
Here I used `cpus = list(os.sched_getaffinity(0))` in the `_resources.py` for cpu allocation.

The resulting allocations reported by clusterduck were following:
```
[3, 4, 5, 27, 28, 29]
[0, 1, 2, 24, 25, 26]
[32, 6, 7, 8, 30, 31]
[33, 34, 35, 9, 10, 11]
```

Which, while I actually only requested 5 cpus and got 6, looks like a good allocation of cores
